app/controllers/flow_controller.rb

```diff
@@ -45,7 +55,16 @@ def index
   end
 
   def about
-    'xx'
+  end
+
+  def restrictions
```

Comment:
ActiveRecord does DB request when you call `.each` on it, or if you say `.all`. 

so most of the time it calls it while he is rendering templates, and that is good a practice as well. because of that, I can modify product search on the fly, in before-render while it did not yet happen.

but in the end you are right, I forgot to add paging in a template, which that would fix that :)